### DISCUSSION TITLE
[Question]: Accessing Live View/Recordings from an external network

### DISCUSSION THREAD

**Discussion by Orochikaku** [blakeblackshear/frigate#13131] [Ask A Question]
### What is your question:

I've done a little DIY CCTV setup for my folks at their house and I'm mulling over how best to give them access to the Frigate web UI while they're away from home.

I've read #5415 and [this reddit post](https://www.reddit.com/r/frigate_nvr/comments/159cald/securing_your_frigate_installation_for_remote/) but these posts were both made before 0.14 which now has Authentication and TLS. 

So, my question is, what's the recommended way to enable this remote i.e., over the internet access to the Frigate web UI?

---

- [docker-compose.yml](https://gist.github.com/Orochikaku/3d9eea5147099462aff143ba547a0bac)
- [config.yaml](https://gist.github.com/Orochikaku/f7c66a6bac425bc193bb68ddf0b0bc6a)
- [Hardware](https://lucid.app/lucidchart/f6c7fc39-e502-468d-b965-fda250901d86/view)

Note: I have a non-static IP

---

P.S. I do some software dev but I'm clueless when it comes to security and networking so I'd appreciate if the answers could be dumbed down a bit...

I should be competent enough to setup DDNS and I've already got a domain, and should be competent enough to setup a sub-domain to point to the Frigate web UI.

**Comment by nwish** [comment]
You can do as you plan, but reverse proxy in to Frigate so you don’t have to open any other ports.

**Reply by Orochikaku** [reply]
I would still need to open ports for WebRTC too anyway right?

**Reply by nwish** [reply]
Nope!  Your internal Frigate instance will handle all that.  You just need 443 open if you plan to use a standard HTTPS protocol connection with your domain name.  This is exactly what I do and it works well.  

Domain (HTTPS) via port 443 —> Router port forward to reverse proxy (I use Nginx) —> Reverse proxy translation to Frigate authenticated port (you can change this internally if you want) 

Disable TLS in Frigate config.

**Comment by hawkeye217** [comment]
The official Frigate docs have instructions on setting up a reverse proxy: https://docs.frigate.video/guides/reverse_proxy

**Reply by Orochikaku** [reply]
I understand that the documentation for a reverse proxy exists, but even on that page it doesn't actually make a recommendation amongst the three options there. Further, in #5415 a collaborator, @NickM-27 to the project themselves don't merely use a reverse proxy.

Hence, why I'm asking as someone out of the loop on security best practices, to help me decide.

**Reply by hawkeye217** [reply]
A reverse proxy is certainly a common method.  I personally just use an on-demand VPN (WireGuard), which is dependent on other hardware and setup but can sometimes be more complex for an end user. A cloudflare tunnel is also a popular option. 

We don’t necessarily make a recommendation because all of these are good options.

**Comment by trythatagain** [comment]
Have thought of using [Tailscale](https://tailscale.com/)
No ports need to be open, there are clients for most OS's, plus if you run PFsense then you can also install directly on your router, not sure about OPNsense.
Also if you feel like it you could also selfhost Headscale

**Reply by Orochikaku** [reply]
I was interested in it till I learned that they keep your keys, and that headscale still requires the use of tailscale's proprietary client.

**Reply by trythatagain** [reply]
Fair enough, I'd stick with wiregaurd or openvpn then.

**Comment by GitWally** [comment]
I had a similar problem: solved with Cloudflare tunnel to port 8971 (where authentication is required).
But I had a domain to leverage since I use it to expose HA as well, using Cloudflare. 
Thanks
